lib/src/json_schema/validator.dart

```diff
@@ -760,6 +804,13 @@ class Validator {
     if (!_reportMultipleErrors && _treatWarningsAsErrors) throw FormatException(msg);
   }
 
+  Function(String, String, String) _errFuncForVocabulary(Uri v) {
+    return _vocabulary.contains(v)
+        ? _err
+        : (String msg, String instancePath, String schemaPath) =>
+            _warn('ignoring vocabulary $v: $msg', instancePath, schemaPath);
+  }
+
   JsonSchema _rootSchema;
   List<ValidationError> _errors = [];
   List<ValidationError> _warnings = [];
```

Comment:
We probably need to figure out how to return these to consumers more easily. Currently there is a `validateWithErrors` that returns `List<ValidationError>`.

Reply:
Suppose we could add a new: `ValidationResult validateWithResults()` where:

```
class ValidatationResult {
 final List<ValidationError> errors;
 final List<ValidationError> warnings;
 // TODO: add standard error outputs and annotations later?
}

Reply:
I like that! Probably would be scope creep for this PR though.

Reply:
That would also open the door to returning collected annotation results, FWIW.

Reply:
Although, someone can already call `validator.validate()` and then access the `validator.warnings()` and `validator.errors()` today :man_shrugging:

Reply:
Yeah, we've kept that API around because no need to break it, but it feels a little funky not to get a validation result. It's ok for this PR